homeworks/taridaniel/3_conditionals_loops/rock_paper_scissors.py

```diff
@@ -0,0 +1,43 @@
+while True:
+    rounds = int(input("Enter the number of rounds (must be an odd number): "))
+    if rounds % 2 != 0:
+        break
+    else:
+        print("Error: Please enter an odd number.")
+
+player1_points = 0
+player2_points = 0
+possible_choices = ["rock", "paper", "scissors"]
+
+# Play the game for a certain number of rounds
+for word in range(rounds):
+    while True:
+        player1_choice = input("Player 1, enter your choice (rock, paper, or scissors): ")
+        if player1_choice in possible_choices:
+            break
+        else:
+            print("Error: Invalid choice. Please enter rock, paper, or scissors.")
+
+    while True:
+        player2_choice = input("Player 2, enter your choice (rock, paper, or scissors): ")
+        if player2_choice in possible_choices:
+            break
+        else:
+            print("Error: Invalid choice. Please enter rock, paper, or scissors.")
+
+    if player1_choice == player2_choice:
+        print("It's a tie!")
```

Comment:
Ezzel az a gond, hogy ha mindhárom kör döntetlen, akkor a játék is döntetlen lesz.
A kódod viszont azt írja ki hogy:
`Player 2 wins by 0 points!`

Mert a végén nem kezeled a döntetlen lehetőségét, csak hogy egyiküknek több pontja van e a másiknál (40-43 sorok).
Vagy kezeld le a végén a döntetlent, vagy ha életszerűbbre szeretnéd a játékot, akkor igazából egy kör se térhet véget amíg ugyanazt az input-ot adják. Ezt röviden megoldhatod így:
```

possible_choices = ["rock", "paper", "scissors"]
current_round = 1
# Play the game for a certain number of rounds
while current_round <= rounds:
#for word in range(rounds):
    while True:
        player1_choice = input("Player 1, enter your choice (rock, paper, or scissors): ")
        if player1_choice in possible_choices:
            break
        else:
            print("Error: Invalid choice. Please enter rock, paper, or scissors.")

    while True:
        player2_choice = input("Player 2, enter your choice (rock, paper, or scissors): ")
        if player2_choice in possible_choices:
            break
        else:
            print("Error: Invalid choice. Please enter rock, paper, or scissors.")

    if player1_choice == player2_choice:
        print("It's a tie, try this round again!")
        continue
    elif (player1_choice == "rock" and player2_choice == "scissors") or \
         (player1_choice == "scissors" and player2_choice == "paper") or \
         (player1_choice == "paper" and player2_choice == "rock"):
        player1_points += 1
    else:
        player2_points += 1
    
    current_round += 1

```

ami itt történt: 
1. bevezettem egy current_round változót, ami a körök számát fogja track-elni.
2. a for loopot lecseréltem egy while loopra, addig loop-olunk amíg a current_round ami 1-től kezdődik, eléri a rounds-ot.
3. Ha döntetlent játszanak, akkor a continue-val újrakezdjük a kört. Azért kezdődik újra a kör, mert így nem jut el a kód a loop végére, ahol növelnénk a current_round értékét 1-gyel. Így addig megy minden kör amíg eltérő input-ot nem adnak meg, és nem kell a döntetlent lekezelned a végén egy plusz condition-nel.